Source/WebGPU/WebGPU/ShaderModule.mm

```diff
@@ -93,8 +99,14 @@
         auto convertedPipelineLayout = ShaderModule::convertPipelineLayout(WebGPU::fromAPI(hint.hint.layout));
         wgslHints.add(hintKey, WTFMove(convertedPipelineLayout));
     }
+
     auto prepareResult = WGSL::prepare(std::get<WGSL::SuccessfulCheck>(checkResult).ast, wgslHints);
-    auto library = ShaderModule::createLibrary(device.device(), prepareResult.msl, WTFMove(label));
+#if ENABLE(WEBGPU)
```

Comment:
@litherum are you ok with this to avoid checking in the metal source file?

Even once we have the compiler outputting a basic shader, it is likely there will be disconnects between the pipeline and compiler where the pipeline may want to test features not available in the compiler. Having an option to use custom metal source allows development on the pipeline to continue in parallel with the compiler.

Reply:
I don't quite understand the ENABLE(WEBGPU) check, because this is inside WebGPU.framework. It doesn't make a lot of sense to say "build WebGPU.framework but disable WebGPU"

I'm a little sad that this is necessary, but I do see the rationale why it is significantly helpful for development. Can we at least use like

```
#define TEMPORARY_MSL_HACK_PLEASE_DELETE_ONCE_WGSL_COMPILER_IS_HOOKED_UP 1
```

or something?